collector/lib/ConfigLoader.cpp

```diff
@@ -488,9 +480,9 @@ ConfigLoader::Result ConfigLoader::LoadConfiguration(const std::optional<const Y
     errors = parser_.Parse(&runtime_config, *node);
   }
 
-  if (errors) {
+  if (!errors.empty()) {
```

Comment:
I don't really like this, `if (errors) {` is a much more idiomatic way to express that you will execute something if errors happened when compared to `if (!errors.empty()) {`. Performance wise, both options should be the same, because we create vectors in the inner methods, so not much difference there and this code is not in any hot path, so I wouldn't worry about it.